pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotHelixTaskResourceManager.java

```diff
@@ -427,10 +491,134 @@ private static String getTaskType(String name) {
     return name.split(TASK_NAME_SEPARATOR)[1];
   }
 
+  public static class TaskDebugInfo {
+    private String _startTime = null;
+    private String _executionStartTime = null;
+    private TaskState _taskState = null;
+    private TaskCount _subTaskCounts = null;
+    private List<TaskPartitionDebugInfo> _subTaskInfos = null;
+
+    public TaskDebugInfo() {
+    }
+
+    public void setStartTime(String startTime) {
+      _startTime = startTime;
+    }
+
+    public void setExecutionStartTime(String executionStartTime) {
+      _executionStartTime = executionStartTime;
+    }
+
+    public void setTaskState(TaskState taskState) {
+      _taskState = taskState;
+    }
+
+    public void setSubTaskCount(TaskCount subTaskCount) {
+      _subTaskCounts = subTaskCount;
+    }
+
+    public void addSubTaskInfo(TaskPartitionDebugInfo subTaskInfo) {
+      if (_subTaskInfos == null) {
+        _subTaskInfos = new ArrayList<>();
+      }
+      _subTaskInfos.add(subTaskInfo);
+    }
+
+    public String getStartTime() {
+      return _startTime;
+    }
+
+    public String getExecutionStartTime() {
+      return _executionStartTime;
+    }
+
+    public TaskState getTaskState() {
+      return _taskState;
+    }
+
+    public TaskCount getSubTaskCount() {
+      return _subTaskCounts;
+    }
+
+    public List<TaskPartitionDebugInfo> getSubTaskInfos() {
+      return _subTaskInfos;
+    }
+  }
+
+  public static class TaskPartitionDebugInfo {
+    private String _taskId = null;
```

Comment:
(nit) no need to explicitly put `= null` for member variables

Reply:
cleaned.